### PR TITLE
fix(gatsby-starter-blog): Move hardcoded author summary to site metadata (#21747)

### LINKED ISSUE #21747: blog-starter template has an hardcoded author summary
## Description

The blog starter template is a great way to have a blog site up and running, but while most of the stuff can be configured from the `gatsby-config.js` file, the author summary is missing and is hardcoded in `src/components/bio.js`. 

### Steps to reproduce

1. `gatsby new gatsby-starter-blog https://github.com/gatsbyjs/gatsby-starter-blog`
2. Update your details in `gatsby-config.js`.
3. Run `gatsby develop` and watch your changes in the browser.

### Actual result

Note the sub-title of the site: 
> Written by `John Doe` who lives and works in San Francisco building useful things. You should follow him on Twitter

The `John Doe` part can be changed from the default `Kyle Mathews`, but it is difficult to change its summary without changing the `bio.js` file.

### Expected result

Some of us live out of San Francisco. :smile: 

### Environment

```
  System:
    OS: Linux 4.19 Debian GNU/Linux 10 (buster) 10 (buster)
    CPU: (8) x64 Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
    Shell: 5.7.1 - /usr/bin/zsh
  Binaries:
    Node: 10.15.2 - /usr/bin/node
    npm: 5.8.0 - /usr/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 80.0.3987.116
    Firefox: 68.5.0esr
  npmPackages:
    gatsby: ^2.19.7 => 2.19.7 
    gatsby-image: ^2.2.39 => 2.2.39 
    gatsby-plugin-feed: ^2.3.26 => 2.3.26 
    gatsby-plugin-google-analytics: ^2.1.34 => 2.1.34 
    gatsby-plugin-manifest: ^2.2.39 => 2.2.39 
    gatsby-plugin-offline: ^3.0.32 => 3.0.32 
    gatsby-plugin-react-helmet: ^3.1.21 => 3.1.21 
    gatsby-plugin-sharp: ^2.4.3 => 2.4.3 
    gatsby-plugin-typography: ^2.3.21 => 2.3.21 
    gatsby-remark-copy-linked-files: ^2.1.36 => 2.1.36 
    gatsby-remark-images: ^3.1.42 => 3.1.42 
    gatsby-remark-prismjs: ^3.3.30 => 3.3.30 
    gatsby-remark-responsive-iframe: ^2.2.31 => 2.2.31 
    gatsby-remark-smartypants: ^2.1.20 => 2.1.20 
    gatsby-source-filesystem: ^2.1.46 => 2.1.46 
    gatsby-transformer-remark: ^2.6.48 => 2.6.48 
    gatsby-transformer-sharp: ^2.3.13 => 2.3.13 
```

### PR DESCRIPTION
**Pull request by tomer**
## Description

The blog starter template is a great way to have a blog site up and running, but while most of the stuff can be configured from the `gatsby-config.js` file, the author summary is missing and is hardcoded in `src/components/bio.js`.

## Related Issues
Fixes  #21747

### CODE REVIEW

File: starters/blog/src/components/seo.js [hunk 1]
```diff
@@ -18,7 +18,6 @@ const SEO = ({ description, lang, meta, title }) => {
           siteMetadata {
             title
             description
-            author
```
Comment (vladar) [inline starters/blog/src/components/seo.js]: Shouldn't be deleted. There is one usage here:

https://github.com/gatsbyjs/gatsby/blob/b25b52b51746831d051443fc7c66f682eacf7f6e/starters/blog/src/components/seo.js#L57-L60

Instead, query for `{ author { name } }` and replace the usage above with `site.siteMetadata.author.name`

File: starters/blog/src/components/bio.js [hunk 2]
```diff
@@ -23,7 +23,10 @@ const Bio = () => {
       }
       site {
         siteMetadata {
-          author
+          author {
+            name
+            summary
+          }
```
Comment (vladar) [inline starters/blog/src/components/bio.js]: There is one more usage of `author` in this file:

https://github.com/gatsbyjs/gatsby/blob/b25b52b51746831d051443fc7c66f682eacf7f6e/starters/blog/src/components/bio.js#L46-L48

Should be also changed to `alt={author.name}`

**Comment by tomer** [comment]
> Thanks for the PR! Added comments inline with requested changes.

Thanks for the suggestions, I'll include the suggested changes right away. What is the prefered way of merging commits? Should I run `amend` on the current commit or make the changes in separeted commit?

File: starters/blog/src/components/seo.js (hunk 1 above)
Comment (tomer) [reply starters/blog/src/components/seo.js]: Well, I afraid that this is another bug. Should be `site.siteMetadata.social.twitter` as it used as attribution to the Twitter handle of the publisher. → #21773 

https://developer.twitter.com/en/docs/tweets/optimize-with-cards/guides/getting-started#card-and-content-attribution

**Comment by smthomas** [comment]
After reviewing this and the code in #21774, it would be best to merge these into a single PR. Everything looks good but can you merge the code from #21774 into this PR with an additional commit and then close 21774?

Once that is ready we can get this merged. Thanks for your contribution.

**Comment by tomer** [comment]
> merge the code from #21774 into this PR with an additional commit

Done with commit 5293b14173f7224859b93634ae99337943bec7c5

**Pull request merged**